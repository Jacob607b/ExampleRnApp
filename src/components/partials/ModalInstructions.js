import React from "react";
import { View, Text, Modal, TouchableOpacity } from "react-native";
import { Button } from "react-native-paper";

const ModalInstructions = () => {
  const [visible, setVisible] = React.useState(true);
  const hideModal = () => setVisible(false);
  const containerStyle = {
    backgroundColor: "white",
    padding: 20,
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    textAlign: "center",
    boderRadius: 10,
    width: "100%"
  };

  return (
    <Modal
      visible={visible}
      onRequestClose={hideModal}
      transparent={true}
      animationType="slide"
    >
      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
        <View style={containerStyle}>
          <Text style={{ fontWeight: "bold" }}>
            Please select or take all necessary photos before you press "Submit
            Image"
          </Text>
          <TouchableOpacity onPress={hideModal} testID="dismiss">
            <Text>
              <Button mode="elevated">Dismiss</Button>
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};

export default ModalInstructions;
