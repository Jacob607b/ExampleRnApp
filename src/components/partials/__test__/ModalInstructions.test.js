import ModalInstructions from "../ModalInstructions";
import { fireEvent, render, screen } from "@testing-library/react-native";

describe('Modal Instructions', () => {
    test("Modal Instructions Snapshot", async () => {
 
        render(<ModalInstructions />)
    
        expect(screen).toMatchSnapshot();
      });
      test("Modal Instructions Dismiss", () => {
        render(<ModalInstructions />)
        const dismissButton = screen.getByTestId("dismiss");
        expect(dismissButton).toBeOnTheScreen()
        fireEvent.press(dismissButton)
        
      });
   
 })

