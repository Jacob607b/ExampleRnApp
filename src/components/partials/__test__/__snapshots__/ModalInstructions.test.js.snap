// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Instructions Modal Instructions Snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "boderRadius": 10,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 20,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Please select or take all necessary photos before you press "Submit Image"
      </Text>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="dismiss"
      >
        <Text>
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "rgb(247, 243, 249)",
                "borderRadius": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 3,
              }
            }
            testID="button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(247, 243, 249)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "flex": undefined,
                  "minWidth": 64,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
              testID="button-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            Object {
                              "color": "rgba(103, 80, 164, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                    testID="button-text"
                  >
                    Dismiss
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
